benchmark/synthetic_benchmark.py

```diff
@@ -0,0 +1,160 @@
+"""
+This script is modified from the horovod benchmark script(https://github.com/horovod/horovod/blob/master/examples/pytorch/pytorch_synthetic_benchmark.py).
+You can compare the performance with the horovod script directly.
+"""
+
+import argparse
+import timeit
+import torch.backends.cudnn as cudnn
+import torch.nn.functional as F
+import torch.optim as optim
+import torch.utils.data.distributed
+from torchvision import models
+import numpy as np
+import bagua.torch_api as bagua
+
+
+# Benchmark settings
+parser = argparse.ArgumentParser(
+    description="PyTorch Synthetic Benchmark",
+    formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+)
+parser.add_argument(
+    "--fp16-allreduce",
+    action="store_true",
+    default=False,
+    help="use fp16 compression during allreduce",
+)
+
+parser.add_argument("--model", type=str, default="resnet50",
+                    help="model to benchmark")
+parser.add_argument("--batch-size", type=int,
+                    default=32, help="input batch size")
+
+parser.add_argument(
+    "--num-warmup-batches",
+    type=int,
+    default=10,
+    help="number of warm-up batches that don't count towards benchmark",
+)
+parser.add_argument(
+    "--num-batches-per-iter",
+    type=int,
+    default=10,
+    help="number of batches per benchmark iteration",
+)
+parser.add_argument(
+    "--num-iters", type=int, default=10, help="number of benchmark iterations"
+)
+
+parser.add_argument(
+    "--no-cuda", action="store_true", default=False, help="disables CUDA training"
```

Comment:
remove unused args

Reply:
Our aluminum supports the no-cuda mode, I think it should be retained?

Reply:
bagua has not yet wired Al's MPI interface, so not supported (at the moment)

Reply:
OK, I will fix it.

Reply:
also better to add option to fix seeds, so that we can use it for checking numerical correctness (like our old CI)